kotest-framework/kotest-framework-engine/src/commonMain/kotlin/io/kotest/engine/listener/TeamCityTestEngineListener.kt

```diff
@@ -143,7 +143,13 @@ class TeamCityTestEngineListener(
             }
             startTest(testCase)
          }
-         TestType.Dynamic -> Unit
+         TestType.Dynamic -> {
+            val p = testCase.parent
+            // we might have a dynamic inside another dynamic parent, in which case we need to start the dynamic parent
+            if (p != null && p.type == TestType.Dynamic) {
+               if (!started.contains(p.descriptor)) startTestSuite(p)
+            }
+         }
```

Comment:
Should there be a `startTest` here as well?

Reply:
No, with dynamics it's done later because you don't know if a dynamic is a test suite or a test.